vpr/src/place/place_util.cpp

```diff
@@ -427,3 +427,38 @@ void alloc_and_load_legal_placement_locations(std::vector<std::vector<std::vecto
     //avoid any memory waste
     legal_pos.shrink_to_fit();
 }
+
+void set_block_location(ClusterBlockId blk_id, const t_pl_loc& location) {
+    auto& place_ctx = g_vpr_ctx.mutable_placement();
+    auto& device_ctx = g_vpr_ctx.device();
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+
+    std::string block_name = cluster_ctx.clb_nlist.block_name(blk_id);
```

Comment:
Is this routine speed critical (used in try_swap)?  If so, worth noting that in a comment.

Reply:
Faster to get a reference to block_name via
const std::string& block_name = ...
Avoids a string copy, which is a good idea if this routine could ever be called in a speed-critical routine (and it might be, if try_swap etc. use it).  Better style, so I'd change this even if this routine isn't currently called in speed critical routines.

Reply:
Not currently called by try_swap, but I changed code to const std::string& block_name as suggested.